mbed_tools/cli.py

```diff
@@ -1,15 +1,51 @@
 """Integration point with all sub-packages."""
+import logging
+
 import click
+
+from mbed_tools_lib.logging import log_exception, set_log_level
+from mbed_tools_lib.exceptions import ToolsError
+
 from mbed_devices.mbed_tools import cli as mbed_devices_cli
 
 
 CONTEXT_SETTINGS = dict(help_option_names=["-h", "--help"])
+LOGGER = logging.getLogger(__name__)
+
+
+class GroupWithExceptionHandling(click.Group):
+    """A click.Group which handles ToolsErrors and logging."""
+
+    def invoke(self, context: click.Context) -> None:
+        """Invoke the command group.
+
+        Args:
+            context: The current click context.
+        """
+        try:
+            super().invoke(context)
+        except ToolsError as err:
+            traceback = context.params["traceback"]
+            if LOGGER.level != logging.DEBUG and not traceback:
+                err = (
+                    f"{err} Increase the verbosity level to `-vvv` to see debug information."
```

Comment:
Would you be able to provide an example error output in the cli? I wonder if this line is readable as it currently stands. Does this rely on every `{err}` ending with a dot? 🤔

Reply:
Example output
```
ERROR: Failed to find a connected device. Increase the verbosity level to `-vvv` to see debug information. Use the `--traceback` argument to see a full stack trace.
```

Yes it relies on people writing sentences that end with a full stop. Otherwise we get `{err}.. Increase the...` sometimes, which looks silly.

Reply:
Awesome, thanks. Do you think separating that output into two lines would improve things or not? I.e. f"{err}\nIncrease..". Wonder if that slight change would make this more sentence proof. :)

Reply:
Yeah, sounds good.

Reply:
436857a